spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractPollingEndpoint.java

```diff
@@ -266,7 +267,17 @@ private boolean doPoll() {
 			if (holder != null) {
 				holder.setMessage(message);
 			}
-			this.handleMessage(message);
+			try {
+				this.handleMessage(message);
+			}
+			catch (Exception e) {
+				if (e instanceof MessagingException) {
+					throw new OriginalMessageContainingMessagingException(message, (MessagingException) e);
+				}
+				else {
+					throw new MessagingException(message, e);
+				}
```

Comment:
>Is it just a Marker interface? 

Kind of; it's so we know to unwrap the original exception and populate the enhanced error message. But I see you're point, it could be a simple subclass of ME and use the failedMessage property (otherwise it doesn't need to be an ME).

Fixing...